### PR TITLE
Fix Difficulties folder 

### PR DESCRIPTION
**Pull request by TPGPL**

### CODE REVIEW

File: wiki/Difficulties/en.md [hunk 1]
```diff
@@ -13,10 +15,10 @@ Star Rating and Web Icons
 
 This is the range that decides which Web Icons are used depending on the star rating of any given map. Following the icon range is a good estimate to achieve a balanced difficulty spread. There are some guidelines you can follow to improve the difficulty spread of your mapset.
 
-[Standard](Standard) difficulty spreads
+[Standard](/wiki/Game_Modes/osu!) difficulty spreads
```
Comment (MegaApplePi) [inline wiki/Difficulties/en.md]: According to the ASG, section headings are not to have links.
Change to `osu! difficulty spreads` (or `osu!standard difficulty spreads`)

File: wiki/Difficulties/en.md [hunk 2]
```diff
@@ -47,7 +49,7 @@ A good difficulty spread is **more than just Star Rating and Web Icons**! Howeve
 Example Taiko Spreads
```
Comment (MegaApplePi) [inline wiki/Difficulties/en.md]: To be consistent, change to `osu!taiko difficulty spreads`

File: wiki/Difficulties/en.md [hunk 3]
```diff
@@ -13,10 +15,10 @@ Star Rating and Web Icons
 
 This is the range that decides which Web Icons are used depending on the star rating of any given map. Following the icon range is a good estimate to achieve a balanced difficulty spread. There are some guidelines you can follow to improve the difficulty spread of your mapset.
```
Comment (MegaApplePi) [inline wiki/Difficulties/en.md]: change `Web Icons` to `web icons`

File: wiki/Difficulties/en.md [hunk 4]
```diff
@@ -13,10 +15,10 @@ Star Rating and Web Icons
...
 
-[Standard](Standard) difficulty spreads
+[Standard](/wiki/Game_Modes/osu!) difficulty spreads
 --------------------------------------------------
 
-From the [Ranking Criteria](Ranking_Criteria):
+From the [Ranking Criteria](/wiki/Ranking_Criteria):
 
 "**The difficulties in the mapset must be in a consecutive order. Easy or Normal can be skipped if the gap in the star rating spread allows it. If your mapset has two difficulties, one of them cannot be an Insane or Expert. The lowest difficulty must be below 2.0 stars.**"
```
Comment (MegaApplePi) [inline wiki/Difficulties/en.md]: line 37: change `**NOT**` to `**not**`

line 47: change `Star Rating` to `star rating`
line 47: change `Web Icons` to `web icons`

File: wiki/Difficulties/en.md [hunk 5]
```diff
@@ -146,4 +148,4 @@ Rest moments in Taiko
 -   ![Hard](/wiki/shared/Hard-t.png "Hard") Main snapping 1/1 + 1/2 with 3/2 or 2/1 breaks
 -   ![Insane](/wiki/shared/Insane-t.png "Insane") Main snapping 1/2 + 1/4 with 1/1 breaks
 
-For skilled mappers / players who are having problems to distinguish when a lower difficulty becomes too hard, you can try playing your beatmap on **[T-Aiko!](T-Aiko! "wikilink")** which gives you a different difficulty feeling and can show how lower player could feel when playing your maps.
+For skilled mappers / players who are having problems to distinguish when a lower difficulty becomes too hard, you can try playing your beatmap on **[T-Aiko!](/wiki/Game_Modes/External_Ports/T-Aiko/Old)** which gives you a different difficulty feeling and can show how lower player could feel when playing your maps.
```
Comment (MegaApplePi) [inline wiki/Difficulties/en.md]: change `mappers / players` to `mappers or players`

File: wiki/Difficulties/osu!/Easy/en.md [hunk 6]
```diff
@@ -9,34 +12,34 @@ And try to avoid these:
 
 -   stacked notes
 -   overlapping notes
--   [Kick slider](Kick_slider)
+-   Kick slider
```
Comment (MegaApplePi) [inline wiki/Difficulties/osu!/Easy/en.md]: idk if you want to (or if it'll be _set-in-stone_) but `Short Repeating Sliders` are also known as the `Kick Slider`... the soon-to-be link will be `[Short Repeating Sliders](/wiki/Mapping_Techniques/Slider/Short_Repeating_Sliders/)`

File: wiki/Difficulties/osu!/Easy/en.md [hunk 7]
```diff
@@ -9,34 +12,34 @@ And try to avoid these:
...
+You'll want to use [Distance Snapping](/wiki/Beatmap_Editor/Distance_Snap) to allow equal distances over time. So, **adjust the distance snap multiplier so that hit circles only one beat apart don't overlap each other.** For a circle size of 4, a distance snap of 0.5 should work. New players often find hit circles that overlap each other like that hard to read. Most of the time, just having them next to each other without overlapping is good enough.
 
 | This is easier to follow                                | than something like this                                 |
 |---------------------------------------------------------|----------------------------------------------------------|
 | ![Farther Aparts](Farther_Aparts.png "Farther Aparts")  | ![Less so](Less_so.png "Less so")                        |
```
Comment (MegaApplePi) [inline wiki/Difficulties/osu!/Easy/en.md]: From ASG, **never** put images inside tables!

File: wiki/Difficulties/osu!/Easy/en.md [hunk 8]
```diff
@@ -9,34 +12,34 @@ And try to avoid these:
...
+Now, your main rhythm here should be every other beat. Start with the [downbeat](/wiki/Beatmap_Editor/Timing) and start off by placing a hit circle every other beat. However, you'll find that this is boring and puts you to sleep. This is why we go back and edit things to spruce things up a bit so that players don't totally die of boredom. You're going to want to do three things:
 
 Being on the beat
 -----------------
 
-Assume that a "." (period) is a beat without a [hit circle](Hit_Object), and an "O" ("o", not a zero) is a beat with a hit circle (Easy difficulties should be mapped to the beat for the most part, with a few exceptions). Here's a simple pattern you can try:
+Assume that a "." (period) is a beat without a [hit circle](/wiki/Hit_Objects), and an "O" ("o", not a zero) is a beat with a hit circle (Easy difficulties should be mapped to the beat for the most part, with a few exceptions). Here's a simple pattern you can try:
```
Comment (MegaApplePi) [inline wiki/Difficulties/osu!/Easy/en.md]: might as well put code in that period and O
since the font for code is monospace, you could remove the note that it is an "o", not a zero

File: wiki/Difficulties/osu!/Easy/en.md [hunk 9]
```diff
@@ -9,34 +12,34 @@ And try to avoid these:
...
 
 O.O.OOO.
```
Comment (MegaApplePi) [inline wiki/Difficulties/osu!/Easy/en.md]: Maybe an image?

File: wiki/Difficulties/osu!/Easy/en.md [hunk 10]
```diff
@@ -9,34 +12,34 @@ And try to avoid these:
...
```
Comment (MegaApplePi) [inline wiki/Difficulties/osu!/Easy/en.md]: line 48: maybe an image?
line 50: maybe an image?
line 52: maybe an image?

File: wiki/Difficulties/osu!/Easy/en.md [hunk 11]
```diff
@@ -60,7 +63,7 @@ Avoid using sliders that only use 1/2 of a beat in the timeline. The rhythm thes
 Being visually appealing
 ------------------------
 
-Since the map is considered to be easy, it is recommended to make the map visually appealing to look at because patterns will definitely be noticed and appreciated on an Easy. This can be achieved through [smooth looking curves](Making_Good_Sliders) and strong structured patterns.
+Since the map is considered to be easy, it is recommended to make the map visually appealing to look at because patterns will definitely be noticed and appreciated on an Easy. This can be achieved through [smooth looking curves](/wiki/Mapping_Techniques/Making_Good_Sliders) and strong structured patterns.
 
 There are more advanced things you can do with an Easy, so experiment and figure out what works best for you.
```
Comment (MegaApplePi) [inline wiki/Difficulties/osu!/Easy/en.md]: line 77: the streams link is being merged with Rhythm ... I'm not sure if it will be final, but it will be `[Streams](/wiki/Mapping_Techniques/Rhythm)`

----
If you think I should undo this change, let me know.

File: wiki/Difficulties/osu!/Expert/en.md [hunk 12]
```diff
@@ -1,19 +1,22 @@
-In ways to push and extend the limits of beatmapping, Expert was invented/created to be beatmaps *like* [Insane](Insane) but go beyond.
+Expert
+========
+
+In ways to push and extend the limits of beatmapping, Expert was invented/created to be beatmaps *like* [Insane](/wiki/Difficulties/osu!/Insane) but go beyond.
 
 It could be the extra *5th* difficulty (if beatmaps are formatted in ![Easy](/wiki/shared/Easy-s.png "Easy")![Normal](/wiki/shared/Normal-s.png "Normal")![Hard](/wiki/shared/Hard-s.png "Hard")![Insane](/wiki/shared/Insane-s.png "Insane")![Expert](/wiki/shared/Expert-s.png "Expert")) or the *4th* (if beatmaps are formatted in ![Easy](/wiki/shared/Easy-s.png "Easy")![Normal](/wiki/shared/Normal-s.png "Normal")![Hard](/wiki/shared/Hard-s.png "Hard")![Insane](/wiki/shared/Insane-s.png "Insane")![Expert](/wiki/shared/Expert-s.png "Expert")). These maps, like [Insane](Insane), quote, "generally exemplifications of the mapper's skill, and are the difficulties where creativity can be easily shown."
```
Comment (MegaApplePi) [inline wiki/Difficulties/osu!/Expert/en.md]: the Insane link will not work (last sentence of this line)

File: wiki/Difficulties/osu!/Expert/en.md [hunk 13]
```diff
@@ -1,19 +1,22 @@
...
-An issue with Expert maps, like Insane maps, is that they can easily be [Overmapped](Overmapped) to increase the map's difficulty.
+An issue with Expert maps, like Insane maps, is that they can easily be Overmapped to increase the map's difficulty.
 
 Settings
 --------
 
 *Note: this is only a general suggestion. The difficulty setting will vary depending on the song itself and mapper's own style.*
 
-Also note: in most cases it's the [mapping](mapping), not the [Song Setup](Song_Setup), but doing this (or similar) is a step of the way!''
+Also note: in most cases it's the [mapping](/wiki/Beatmapping), not the [Song Setup](/wiki/Beatmap_Editor/Song_Setup), but doing this (or similar) is a step of the way!''
```
Comment (MegaApplePi) [inline wiki/Difficulties/osu!/Expert/en.md]: remove line?

File: wiki/Difficulties/osu!/Hard/en.md [hunk 14]
```diff
@@ -1,9 +1,12 @@
...
 
-These maps generally have a higher [approach rate](approach_rate), [HP drain](HP_drain) rate and [overall difficulty](overall_difficulty) compared with [Normal](Normal) difficulties. More complex/difficult patterns also appear in this difficulty such as short [streams](streams) and [jumps](jumps). Depending on the mapper's ability and the song used, this name will be used instead of Insane, based on mapper's relative comparison on testplay results.
+These maps generally have a higher [Approach Rate](/wiki/Beatmap_Edtior/Song_Setup), [HP Drain](/wiki/Beatmap_Edtior/Song_Setup) rate and [Overall Difficulty](/wiki/Beatmap_Edtior/Song_Setup) compared with [Normal](/wiki/Difficulties/osu!/Normal) difficulties. More complex/difficult patterns also appear in this difficulty such as short [streams](/wiki/Mapping_Techniques/Streams) and [jumps](/wiki/Mapping_Techniques/Jumps). Depending on the mapper's ability and the song used, this name will be used instead of Insane, based on mapper's relative comparison on testplay results.
```
Comment (MegaApplePi) [inline wiki/Difficulties/osu!/Hard/en.md]: (stating again)
the streams link is being merged with Rhythm ... I'm not sure if it will be final, but it will be `[Streams](/wiki/Mapping_Techniques/Rhythm)`

----
If you think I should undo this change, let me know.

Comment (MegaApplePi) [inline wiki/Difficulties/osu!/Hard/en.md]: the jumps link is changed to `[jumps](/wiki/Mapping_Techniques/Jump)`

File: wiki/Difficulties/osu!/Hard/en.md [hunk 15]
```diff
@@ -1,9 +1,12 @@
...
 
 If your map has a Hard difficulty you should try to keep the map considerably easier than the Insane. (Avoid big jumps and fast/long streams)
```
Comment (MegaApplePi) [inline wiki/Difficulties/osu!/Hard/en.md]: change to (just moved things around)
`If your map has a Hard difficulty you should try to keep the map considerably easier than the Insane (avoid big jumps and fast/long streams).`

File: wiki/Difficulties/osu!/Hard/en.md [hunk 16]
```diff
@@ -21,7 +24,7 @@ Difficulty setup should include a BPM appropriate slider velocity that captures
 And try to avoid these:
 
 -   extremely long streams
--   fullscreen jump patterns ([Jumps](Jumps) that are wide as the screen)
+-   fullscreen jump patterns ([Jumps](/wiki/Mapping_Techniques/Jumps) that are wide as the screen)
```
Comment (MegaApplePi) [inline wiki/Difficulties/osu!/Hard/en.md]: (stating again)
`the jumps link is changed to [jumps](/wiki/Mapping_Techniques/Jump)`

File: wiki/Difficulties/osu!/Normal/en.md [hunk 17]
```diff
@@ -29,25 +32,25 @@ Setup
...
 
-Remember, think of simple [patterns](Mapping_Techniques#Patterns) and note placements that everyone would be able to hit and identify, no matter who plays your normal difficulty^. Remember to use [sliders](Hit_Objects#Sliders) whenever a long note or key is held in the song you are mapping. If the part of the map you're mapping plays better with a slider, add it in there too. You can never make enough awesome sliders in Normal and Easy difficulties^^.
+Remember, think of simple [patterns](/wiki/Mapping_Techniques) and note placements that everyone would be able to hit and identify, no matter who plays your normal difficulty^. Remember to use [sliders](/wiki/Hit_Objects) whenever a long note or key is held in the song you are mapping. If the part of the map you're mapping plays better with a slider, add it in there too. You can never make enough awesome sliders in Normal and Easy difficulties^^.
```
Comment (MegaApplePi) [inline wiki/Difficulties/osu!/Normal/en.md]: I'd use superscripts .-. (as follows in this line, 54 and 56)

File: wiki/Difficulties/osu!taiko/Kantan/en.md [hunk 18]
```diff
@@ -1,4 +1,7 @@
...
+Kantan
+==========
+
+A **Kantan** difficulty is, in general, the entry point for new players who come for the first time with the game [Taiko](/wiki/Game_Modes/osu!taiko) in touch. That means, these people are not familiar with the game and need to learn the basics of it: **How to hit the notes** and facing for the first time **easy pattern constellations**. To achieve this, you have to map mostly(not only) independently of the songs beat or vocal, since these are often used uneven which the beginner can't handle well enough. In that meaning: Typical for a Kantan is the usage of 1/1 notes and 2/1 spaces or above, which gives the new player the possibility to prepare himself for the next patterns.
```
Comment (MegaApplePi) [inline wiki/Difficulties/osu!taiko/Kantan/en.md]: add a space here `mostly(not only)` (change to `mostly (not only)`)

File: wiki/Difficulties/osu!taiko/Kantan/en.md [hunk 19]
```diff
@@ -1,4 +1,7 @@
...
 
 Guidelines
 ----------
```
Comment (MegaApplePi) [inline wiki/Difficulties/osu!taiko/Kantan/en.md]: line 11: change `mostly(not only)`  to `mostly (not only)`
line 15: short spinners is an item on the Mapping Techniques list; link to it `[Short Spinners](/wiki/Mapping_Techniques/Spinner/Short_Spinners)`

File: wiki/Difficulties/osu!taiko/Muzukashii/en.md [hunk 20]
```diff
@@ -1,4 +1,7 @@
-A **Muzukashii** difficulty is now for those who got used to [Taiko](Taiko); who know how to handle basic patterns and different spacings. These properties from **Kantan** and **Futsuu** appear in a Muzukashii as well, but in a large amount. - **Patterns can become longer and spacings more infrequent.** Additional to this, a Muzukashii can also provide **short 1/4, 1/3 and 1/6 patterns** which structure is preferable monotonically and short. The rule of thumb is the following one:
+Muzukashii
+=============
+
+A **Muzukashii** difficulty is now for those who got used to [Taiko](/wiki/Game_Modes/osu!taiko); who know how to handle basic patterns and different spacings. These properties from **Kantan** and **Futsuu** appear in a Muzukashii as well, but in a large amount. - **Patterns can become longer and spacings more infrequent.** Additional to this, a Muzukashii can also provide **short 1/4, 1/3 and 1/6 patterns** which structure is preferable monotonically and short. The rule of thumb is the following one:
 
 -   Stick to simple 1/4 patterns like **ddd** / **kkk**, **ddk** / **kkd** but do not overuse them.
```
Comment (MegaApplePi) [inline wiki/Difficulties/osu!taiko/Muzukashii/en.md]: may need to explain what a `k` or a `d` means, or just make an image to represent this (as follows for this line, 7, and 9)

File: wiki/Difficulties/en.md (hunk 5 above)
Comment (sr229) [inline wiki/Difficulties/en.md]: L151: Seems unecessary since you can just request a playtest from players with different skillsets

File: wiki/Difficulties/en.md [hunk 21]
```diff
@@ -1,10 +1,12 @@
+Difficulty appropriate gameplay elements
```
Comment (sr229) [inline wiki/Difficulties/en.md]: L1-L2: Aren't we suppose to use ``#`` for Titles?

Comment (MegaApplePi) [inline wiki/Difficulties/en.md]: the `==========` is also level 1 heading

Comment (sr229) [inline wiki/Difficulties/en.md]: ah

Comment (MegaApplePi) [inline wiki/Difficulties/en.md]: oh, to answer the question, it doesn't matter which style you use

File: wiki/Difficulties/en.md (hunk 2 above)
Comment (sr229) [inline wiki/Difficulties/en.md]: L49: Revise. Suggested ``Example of Taiko Spreads``.

File: wiki/Difficulties/en.md [hunk 22]
```diff
@@ -1,53 +1,55 @@
...
 
-Example Taiko Spreads
+osu!taiko difficulty spreads
 ---------------------
 
-From the [Ranking Criteria](Ranking_Criteria "wikilink"):
+From the [Ranking Criteria](/wiki/Ranking_Criteria):
 
 -   **The difficulties in the mapset must be in a consecutive order.** Easy or Normal can be skipped if the gap in the star rating spread allows it. If your mapset has two difficulties, one of them cannot be an Insane or Expert. The lowest difficulty must be below 2.0 stars.
 -   **A mapset cannot have just one Taiko difficulty.** If you decide to include Taiko in your map, make sure you have at least 2 Taiko mode difficulties and that one of them is of an easier difficulty level if it doesn't have an osu! standard Easy level difficulty.
```
Comment (MegaApplePi) [inline wiki/Difficulties/en.md]: `Taiko` to `osu!taiko` (I will ignore any recursive occurrences)

File: wiki/Difficulties/osu!/Expert/en.md [hunk 23]
```diff
@@ -1,19 +1,22 @@
...
+It could be the extra *5th* difficulty (if beatmaps are formatted in ![Easy icon](/wiki/shared/Easy-s.png "Easy icon")![Normal icon](/wiki/shared/Normal-s.png "Normal icon")![Hard icon](/wiki/shared/Hard-s.png "Hard icon")![Insane icon](/wiki/shared/Insane-s.png "Insane icon")![Expert icon](/wiki/shared/Expert-s.png "Expert icon")) or the *4th* (if beatmaps are formatted in ![Easy icon](/wiki/shared/Easy-s.png "Easy icon")![Normal icon](/wiki/shared/Normal-s.png "Normal icon")![Hard icon](/wiki/shared/Hard-s.png "Hard icon")![Insane icon](/wiki/shared/Insane-s.png "Insane icon")![Expert icon](/wiki/shared/Expert-s.png "Expert icon")). These maps, like [Insane](/wiki/Difficulties/osu!/Insane), quote, "generally exemplifications of the mapper's skill, and are the difficulties where creativity can be easily shown."
+
+On the osu!beatmap listing page (for example: [Rise Against - The Good Left Undone (Extreme)](https://osu.ppy.sh/b/774564)), one may tell it is an Expert map with this icon: ![Expert icon](/wiki/shared/Expert-s.png "Expert icon")
```
Comment (MegaApplePi) [inline wiki/Difficulties/osu!/Expert/en.md]: the beatmap link should actually be:
`[Rise Against - The Good Left Undone (pishifat) \[Extreme\]](https://osu.ppy.sh/b/774564)`

File: wiki/Difficulties/osu!/Normal/en.md [hunk 24]
```diff
@@ -1,9 +1,12 @@
...
 Summary
 -------
 
-Normal [difficulties](difficulty) are difficulties in which you must have basic skills to play them. They are the easiest difficulties next to [Easy](Easy_(Difficulty)) difficulties.
+Normal [difficulties](/wiki/Difficulties) are difficulties in which you must have basic skills to play them. They are the easiest difficulties next to [Easy](/wiki/Difficulties/osu!/Easy) difficulties.
 
-[Mapping](Mapping) good Normal difficulties can appear to be hard at times, but after a while, it's pretty simple. Here's how to get things started.
+[Mapping](/wiki/Beatmapping) good Normal difficulties can appear to be hard at times, but after a while, it's pretty simple. Here's how to get things started.
 
 Guidelines
```
Comment (MegaApplePi) [inline wiki/Difficulties/osu!/Normal/en.md]: I assume these aren't from some kind of Code of Conduct or Ranking **Guidelines**... I suggest to rename this heading to `Mapping Tips` or similar. (I will ignore any recursive occurrences)

**Comment by RaikireHiuduo** [comment]
If there is a link to T_Aiko!, use this link.

`/wiki/Game_Modes/External_Ports/T_Aiko!`

I am recreating the v2 en.md file now (cl8n(?) burned down the profile pictures in WEIRD(?) iirc, which I do not care much). It pretty odd that v1 survive and v2 did not...
